### PR TITLE
Support Apache Kafka output

### PR DESCRIPTION
**Pull request by KeisukeYamashita**
**What type of PR is this?**

> Uncomment one (or more) `/kind <>` lines:

> /kind bug

> /kind cleanup

> /kind design

> /kind documentation

> /kind failing-test

/kind feature

**Any specific area of the project related to this PR?**

> Uncomment one (or more) `/area <>` lines:

> /area build

> /area config

/area outputs

> /area tests

> /area helm

**What this PR does / why we need it**:

[Apache Kafka](https://kafka.apache.org/) is one of the most famous open-source distributed event streaming platform.
I use it a lot so I want falcosidekick to support. My goal is not requesting falcosidekick to support all products. But IMO, it's worth supporting Kafka.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

This is just a suggestion so please feel free to reject.

I tested as below:

```console
# Terminal 1. Wait for the event
$ kafka-console-consumer --bootstrap-server localhost:9092 --topic test --from-beginning

# Terminal 1
$ go run ./ -c /tmp/falcosidekick.yaml 
```

After accessing `/test` endpoint, it can see below

```console
# Terminal 1. Wait for the event
$ kafka-console-consumer --bootstrap-server localhost:9092 --topic test --from-beginning
{"output":"08:12:46.746475122: Noticcce System user ran an interactive command (user=bin user_loginuid=-1 command=login container_id=79de7a68b24a image=falcosecurity/event-generator) k8s.ns=falco k8s.pod=falco-event-generator-f476f6d5d-gwp2p container=79de7a68b24a k8s.ns=falco k8s.pod=falco-event-generator-f476f6d5d-gwp2p container=79de7a68b24a","priority":"Notice","rule":"System user interactive","time":"2020-11-17T08:12:46.746475122Z","output_fields":{"akey":"AValue","bkey":"BValue","ckey":"CValue","container.id":"79de7a68b24a","container.image.repository":"falcosecurity/event-generator","evt.time":1605600766746475000,"k8s.ns.name":"falco","k8s.pod.name":"falco-event-generator-f476f6d5d-gwp2p","proc.cmdline":"login","user.loginuid":-1,"user.name":"bin"}}
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -247,6 +248,12 @@ googlechat:
   # outputformat: "" # all (default), text
   # minimumpriority: "" # minimum priority of event for using this output, order is emergency|alert|critical|error|warning|notice|informational|debug or "" (default)
   messageformat: "Alert : rule *{{ .Rule }}* triggered by user *{{ index .OutputFields \"user.name\" }}*" # a Go template to format Google Chat Text above Attachment, displayed in addition to the output from `GOOGLECHAT_OUTPUTFORMAT`, see [Slack Message Formatting](#slack-message-formatting) in the README for details. If empty, no Text is displayed before Attachment.
+
+kafka:
+  url: "" # Apache Kafka URL (ex: http://kafka). Defaults to port 9092 if no port is specified after the domain.
```
Comment (Issif) [inline README.md:253]: It misses a message to specify it enables the output

File: config_example.yaml [hunk 2]
```diff
@@ -137,3 +137,9 @@ googlechat:
   # outputformat: "" # all (default), text
   # minimumpriority: "" # minimum priority of event for using this output, order is emergency|alert|critical|error|warning|notice|informational|debug or "" (default)
   messageformat: 'Alert : rule *{{ .Rule }}* triggered by user *{{ index .OutputFields "user.name" }}*' # a Go template to format Slack Text above Attachment, displayed in addition to the output from `SLACK_OUTPUTFORMAT`, see [Slack Message Formatting](#slack-message-formatting) in the README for details. If empty, no Text is displayed before Attachment.
+
+kafka:
+  url: "" # Apache Kafka URL (ex: http://kafka). Defaults to port 9092 if no port is specified after the domain.
```
Comment (Issif) [inline config_example.yaml:142]: Same comment about enabling

File: main.go [hunk 3]
```diff
@@ -283,6 +284,16 @@ func init() {
 		}
 	}
 
+	if config.Kafka.URL != "" && config.Kafka.Topic != "" {
+		var err error
+		kafkaClient, err = outputs.NewKafkaClient(config, stats, promStats, statsdClient, dogstatsdClient)
+		if err != nil {
+			config.GCP.Credentials = ""
```
Comment (Issif) [inline main.go:291]:
```suggestion
			config.Kafka.URL = ""
```

File: outputs/client.go [hunk 4]
```diff
@@ -50,6 +51,7 @@ type Client struct {
 	StatsdClient    *statsd.Client
 	DogstatsdClient *statsd.Client
 	GCPTopicClient  *pubsub.Topic
+	KakaProducer    *kafka.Producer
```
Comment (Issif) [inline outputs/client.go:54]:
```suggestion
	KafkaProducer    *kafka.Producer
```

File: outputs/kafka.go [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+	}
+
+	return &Client{
+		OutputType:      "Kafka",
+		Config:          config,
+		Stats:           stats,
+		PromStats:       promStats,
+		StatsdClient:    statsdClient,
+		DogstatsdClient: dogstatsdClient,
+		KakaProducer:    p,
```
Comment (Issif) [inline outputs/kafka.go:27]:
```suggestion
		KafkaProducer:    p,
```

File: config.go [hunk 6]
```diff
@@ -114,6 +114,10 @@ func getConfig() *types.Configuration {
 	v.SetDefault("Googlechat.OutputFormat", "all")
 	v.SetDefault("Googlechat.MessageFormat", "")
 	v.SetDefault("Googlechat.MinimumPriority", "")
+	v.SetDefault("Kafka.URL", "")
+	v.SetDefault("Kafka.Topic", "")
+	v.SetDefault("Karfa.Partition", 0)
```
Comment (Issif) [inline config.go:119]:
```suggestion
	v.SetDefault("Kafka.Partition", 0)
```

File: outputs/kafka.go [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+	}, nil
+}
+
+// KafkaProduce sends a message to a Apach Kafka Topic
+func (c *Client) KafkaProduce(falcopayload types.FalcoPayload) {
+	b, err := json.Marshal(falcopayload)
+	if err != nil {
+		log.Printf("[ERROR] : Kafka - %v - %v\n", "Error while marshaling message", err.Error())
```
Comment (Issif) [inline outputs/kafka.go:35]:
```suggestion
		log.Printf("[ERROR] : Kafka - %v - %v\n", "Error while marshalling message", err.Error())
```

File: outputs/kafka.go [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+		Value: b,
+	}
+
+	if c.Config.Kafka.Partition == 0 {
+		msg.TopicPartition.Partition = kafka.PartitionAny
+	} else {
+		msg.TopicPartition.Partition = c.Config.Kafka.Partition
+	}
+
+	err = c.KakaProducer.Produce(msg, nil)
```
Comment (Issif) [inline outputs/kafka.go:51]:
```suggestion
	err = c.KafkaProducer.Produce(msg, nil)
```

**Review by Issif — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (KeisukeYamashita) [reply README.md]: Thank you, I added it 👍

File: main.go (hunk 3 above)
Comment (KeisukeYamashita) [reply main.go]: My bad, thank you 🙇

**Review by Issif — APPROVED** [review]
❤️

**Pull request merged**